src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CSharpExpressionCompiler.cs

```diff
@@ -4,16 +4,30 @@
 using System.Collections.Immutable;
 using System.Diagnostics;
 using Microsoft.CodeAnalysis.ExpressionEvaluator;
+using Microsoft.VisualStudio.Debugger;
+using Microsoft.VisualStudio.Debugger.CallStack;
 using Microsoft.VisualStudio.Debugger.Clr;
+using Microsoft.VisualStudio.Debugger.ComponentInterfaces;
 using Microsoft.VisualStudio.Debugger.Evaluation;
 
 namespace Microsoft.CodeAnalysis.CSharp.ExpressionEvaluator
 {
     [DkmReportNonFatalWatsonException(ExcludeExceptionType = typeof(NotImplementedException)), DkmContinueCorruptingException]
-    internal sealed class CSharpExpressionCompiler : ExpressionCompiler
+    internal sealed class CSharpExpressionCompiler : ExpressionCompiler, IDkmLanguageFrameDecoder, IDkmLanguageInstructionDecoder
     {
         private static readonly DkmCompilerId s_compilerId = new DkmCompilerId(DkmVendorId.Microsoft, DkmLanguageId.CSharp);
 
+        // Need to support IDkmLanguageFrameDecoder and IDkmLanguageInstructionDecoder
+        // See https://github.com/dotnet/roslyn/issues/22620
+        private CSharpFrameDecoder _frameDecoder;
+        private CSharpLanguageInstructionDecoder _languageInstructionDecoder;
```

Comment:
`readonly`

Reply:
Thank you!

---
In reply to: [144460747](https://github.com/dotnet/roslyn/pull/22681#discussion_r144460747) [](ancestors = 144460747)